x-pack/plugins/infra/public/components/logging/log_text_stream/loading_item_view.tsx

```diff
@@ -28,122 +44,202 @@ export class LogTextStreamLoadingItemView extends React.PureComponent<
-              id="xpack.infra.logs.noAdditionalEntriesFoundText"
-              defaultMessage="No additional entries found"
-            />
-          </ProgressMessage>
-          {onLoadMore ? (
-            <EuiButtonEmpty size="xs" onClick={onLoadMore} iconType="refresh">
-              <FormattedMessage
-                id="xpack.infra.logs.loadAgainButtonLabel"
-                defaultMessage="Load again"
-              />
-            </EuiButtonEmpty>
-          ) : null}
-        </ProgressEntry>
-      );
-    } else {
-      return null;
-    }
+    const shouldShowCta = !hasMore && !isStreaming;
+
+    const extra = (
+      <EuiFlexGroup justifyContent="center" alignItems="center" gutterSize="m">
+        {isLoading || isStreaming ? (
+          <ProgressSpinner kind={isStreaming ? 'streaming' : 'loading'} />
+        ) : shouldShowCta ? (
+          <ProgressCta
+            position={position}
+            onStreamStart={onStreamStart}
+            onExtendRange={onExtendRange}
+            rangeEdge={rangeEdge}
+          />
+        ) : null}
+      </EuiFlexGroup>
+    );
+
+    return (
+      <ProgressEntryWrapper className={className} position={position}>
+        {position === 'start' ? <>{extra}</> : null}
+        <ProgressMessage timestamp={timestamp} />
+        {position === 'end' ? <>{extra}</> : null}
+      </ProgressEntryWrapper>
+    );
   }
 }
 
-interface ProgressEntryProps {
-  alignment: 'top' | 'bottom';
-  className?: string;
-  color: 'subdued' | 'primary';
-  isLoading: boolean;
-}
+const ProgressEntryWrapper = euiStyled.div<{ position: Position }>`
+  padding-left: ${props => props.theme.eui.euiSizeS};
+  padding-top: ${props =>
+    props.position === 'start' ? props.theme.eui.euiSizeL : props.theme.eui.euiSizeM};
+  padding-bottom: ${props =>
+    props.position === 'end' ? props.theme.eui.euiSizeL : props.theme.eui.euiSizeM};
+`;
+
+const ProgressMessage: React.FC<{ timestamp?: number }> = ({ timestamp }) => {
+  return (
+    <LogTextSeparator>
+      <EuiTitle size="xxs">
+        {timestamp ? (
+          <FormattedMessage
+            id="xpack.infra.logs.showingEntriesUntilTimestamp"
+            defaultMessage="Showing entries until {timestamp}"
+            values={{ timestamp: <FormattedTime value={timestamp} /> }}
+          />
+        ) : (
+          <FormattedMessage
+            id="xpack.infra.logs.noAdditionalEntriesFoundText"
+            defaultMessage="No additional entries found"
+          />
+        )}
+      </EuiTitle>
+    </LogTextSeparator>
+  );
+};
+
+const ProgressSpinner: React.FC<{ kind: 'streaming' | 'loading' }> = ({ kind }) => (
+  <>
+    <EuiFlexItem grow={false}>
+      <EuiLoadingSpinner size="l" />
+    </EuiFlexItem>
+    <EuiFlexItem grow={false}>
+      <EuiText size="s">
+        {kind === 'streaming' ? (
+          <FormattedMessage
+            id="xpack.infra.logs.streamingNewEntriesText"
+            defaultMessage="Streaming new entries"
```

Comment:
There is a label to show until what time we are showing, which corresponds with the first or last visible log entry.

I made a change when streaming, that shows the relative time since the last loaded entry.